src/components/SearchModal/CurrencySearch.tsx

```diff
@@ -347,8 +347,12 @@ export function CurrencySearch({
     fetchFavoriteTokenFromAddress()
   }, [fetchFavoriteTokenFromAddress])
 
+  const fetchingToken = useRef(false)
```

Comment:
Need cleanup this for cases `fetchListTokens`'s dependencies have changed since this might causing new fetch with new deps call cant be call, instead that, we received old deps fetch data. Suggest to add a useEffect here

Reply:
nice catch, updated.